app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt

```diff
@@ -46,33 +48,41 @@ class UserCardsViewModel : ViewModel() {
     )
     val usc = UserCardsRTDB()
 
-    fun onCardButtonClick() {
-        //this code should be in a separate function in UserCardsRTDB so it can be separated from this activitiy's UI
-
-        val cardUID = card1UID
-        val future = CompletableFuture<DataSnapshot>()
-        userCardCollection.child(cardUID).get().addOnSuccessListener {
-            if (it.value == null) {
-                _getMessage.value = "Card $cardUID was not found in your collection"
-                future.completeExceptionally(NoSuchFieldException())
-            } else {
-                future.complete(it)
-            }
-        }.addOnFailureListener {
-            _getMessage.value = "Error getting $cardUID"
-            future.completeExceptionally(it)
-        }
-        //the actual value gotten from the database
-        future.thenAccept {
-            //val card = usc.transformData(it) //turn the retrieved data into a MagicCard object
-            _setMessage.value = "Card $cardUID was succesfully retrieved from your collection"
-        }
+    /**
+     * Add the card to the user's collection
+     */
+    fun onSetButtonClick(card: MagicCard) {
+        usc.addCardToCollection(card)
+        _setMessage.value = "${card.name} was successfully added to your collection"
     }
 
-    fun addCardToCollection(card: MagicCard) {
-        val msg = usc.addCardToCollection(card)
-        _setMessage.value = msg
+    /**
+     * Remove the card from the user's collection
+     */
+    fun onRemoveButtonClick(card: MagicCard) {
+        usc.removeCardFromCollection(card)
+        _removeMessage.value = "${card.name} was successfully removed from your collection"
```

Comment:
The strings should be put in the string.xml file

Reply:
I'll add the xml strings, but since the fragment where this will finally be implemented will be different, ill leave it as is in the viewmodel.